lib/OrdinaryDiffEqCore/src/solve.jl

```diff
@@ -158,19 +157,62 @@ function DiffEqBase.__init(
     else
         _alg = alg
     end
-    f = prob.f
-    p = prob.p
 
-    # Get the control variables
+    use_old_kwargs = haskey(kwargs,:alias_u0) || haskey(kwargs,:alias_du0)
 
-    if alias_u0
+    if use_old_kwargs
+        aliases = ODEAliasSpecifier()
+        if haskey(kwargs, :alias_u0)
+            message = "`alias_u0` keyword argument is deprecated, to set `alias_u0`,
+            please use an ODEAliasSpecifier, e.g. `solve(prob, alias = ODEAliasSpecifier(alias_u0 = true))"
+            Base.depwarn(message, :init)
+            Base.depwarn(message, :solve)
+            aliases = ODEAliasSpecifier(alias_u0 = values(kwargs).alias_u0)
+        else
+            aliases = ODEAliasSpecifier(alias_u0 = nothing)
+        end
+
+        if haskey(kwargs, :alias_du0)
+            message = "`alias_du0` keyword argument is deprecated, to set `alias_du0`,
+            please use an ODEAliasSpecifier, e.g. `solve(prob, alias = ODEAliasSpecifier(alias_du0 = true))"
+            Base.depwarn(message, :init)
+            Base.depwarn(message, :solve)
+            aliases = ODEAliasSpecifier(alias_u0 = aliases.alias_u0, alias_du0 = values(kwargs).alias_du0)
+        else
+            aliases = ODEAliasSpecifier(alias_u0 = aliases.alias_u0, alias_du0 = nothing)
+        end
+        
+        aliases 
+
+    else
+         # If alias isa Bool, all fields of ODEAliases set to alias
+        if alias isa Bool
+            aliases = ODEAliasSpecifier(alias = alias)
+        elseif isnothing(alias) || alias isa ODEAliasSpecifier 
+            aliases = alias
+        end
```

Comment:
Canonicalize this in the DifEqBase part?

Reply:
I'm not sure what you mean. This is here so that if `alias = true` or `alias = false` in `solve` that gets passed to the `ODEAliasSpecifier` constructor so that every variable is aliased / not aliased.

Reply:
I'm not sure what you mean. This is here so that if `alias = true` or `alias = false` in `solve` that gets passed to the `ODEAliasSpecifier` constructor so that every variable is aliased / not aliased.